util/src/main/java/io/kubernetes/client/util/Config.java

```diff
@@ -74,13 +76,17 @@ public static ApiClient fromToken(String url, String token, boolean validateSSL)
   }
 
   public static ApiClient fromConfig(String fileName) throws IOException {
-    KubeConfig config = KubeConfig.loadKubeConfig(new FileReader(fileName)); // TODO UTF-8
+    KubeConfig config =
+        KubeConfig.loadKubeConfig(
+            new BufferedReader(
+                new InputStreamReader(
+                    new FileInputStream(fileName), StandardCharsets.UTF_8.name())));
     config.setFile(new File(fileName));
     return fromConfig(config);
```

Comment:
We should really close the `FileInputStream` here.  Perhaps:

```java
try (FileInputStream fi = new FIleInputStream(fileName)) {
   KubeConfig config = KubeConfig.loadKubeConfig(...);
   return  fromConfig(config);
}
```

Reply:
Yes try-with-resources is correct way of doing this here. Thanks for your input. I will change it.